### PR TITLE
memory: implement delete entry from automatic suggestions

### LINKED ISSUE #6440: Translation Memory Maintenance - Delete Entries
It would be good to be able to maintain the TM in one way or the other.

Perhaps 
- [ ] Search & Replace in TM
- [ ] Search & Delete entries in TM

Or perhpas, as a "simple" (?) starting point, giving the translator the option to delete single entries from the TM when they see the result list in **Automatic Suggestions**. Like perhaps:

- [ ] Delete single entry in **Automatic Suggestions** view:
image

### PR DESCRIPTION
**Pull request by ParthS007**
## Proposed changes

closes #6440 

- Added a Delete entry button for removing translation suggestion from Automatic Suggestions

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [ ] I have squashed my commits into logic units.
- [ ] I have described the changes in the commit messages.

## Other information

Screenshot 2022-07-18 at 11 19 45

### CODE REVIEW

File: weblate/static/editor/full.js [hunk 1]
```diff
@@ -535,6 +561,11 @@
             '<a class="js-copy-save-machinery btn btn-primary">' +
             gettext("Copy and save") +
             "</a>" +
+            "</td>" +
+            "<td>" +
+            '<a class="js-delete-machinery btn btn-danger">' +
+            gettext("Delete entry") +
+            "</a>" +
```
Comment (nijel) [inline weblate/static/editor/full.js:568]: We're able to delete entries only from a single service (Weblate translation memory), so the button should be conditional here.

File: weblate/machinery/views.py [hunk 2]
```diff
@@ -344,7 +344,7 @@ def handle_machinery(request, service, unit, search=None):
     return JsonResponse(data=response)
 
 
-@require_POST
+@require_http_methods(["DELETE", "POST"])
 def translate(request, unit_id, service):
     """AJAX handler for translating."""
     unit = get_object_or_404(Unit, pk=int(unit_id))
```
Comment (nijel) [inline weblate/machinery/views.py:350]: I'm uncertain if it's a good idea to mix this endpoint with removal. Anyway the removal lacks implementation here.

File: weblate/static/editor/full.js (hunk 1 above)
Comment (ParthS007) [reply weblate/static/editor/full.js:568]: okay, I will add the condition.

File: weblate/machinery/views.py (hunk 2 above)
Comment (ParthS007) [reply weblate/machinery/views.py:350]: For the removal part, I asked a couple of question [here](https://github.com/WeblateOrg/weblate/issues/6440#issuecomment-1186991255)

**Pull request closed without merging**